prov/sockets/src/sock_av.c

```diff
@@ -527,7 +527,7 @@ int sock_av_open(struct fid_domain *domain, struct fi_av_attr *attr,
 
 		_av->table_hdr = mmap(NULL, table_sz, PROT_READ | PROT_WRITE,
 					MAP_SHARED, _av->shared_fd, 0);
-		_av->attr.map_addr = _av->table_hdr + sizeof(_av->table_hdr);
+		_av->attr.map_addr = _av->table_hdr + sizeof(struct sock_av_table_hdr);
```

Comment:
There was a bug in <code>map_addr</code> calculation in first patch which was revealed when I tried to fix the Coverity scan issue (thanks to Coverity Scan!). Basically, remote fabric addresses from <code>fi_av_insert</code> needs to be the indices to the av_table where actual addresses (<code>sock_av_addr</code>) are stored. So I added a index array to keep track of that for shared av case and changed <code>map_addr</code> to point to the start of the index array. The new patch is tested with @sayantansur 's shared av test and it works fine along with the other fabtests. @jithinjosepkl can you please review?